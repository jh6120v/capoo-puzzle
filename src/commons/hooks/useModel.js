import React, { useCallback, useState } from 'react';
import {
    Button, ModelContent, ModelFooter, ModelShadow, ModelTitle, ModelWrap
} from '../../styles/model-style';

const useModel = (title, message = null, confirm = null, confirmText = 'Confirm', cancel = null, cancelText = 'Cancel') => {
    const [isShown, setShown] = useState(false);
    const showModal = useCallback(() => setShown(true), []);
    const hideModal = useCallback(() => setShown(false), []);

    const ModelBox = () => (
        <div>
            <ModelWrap>
                <ModelTitle data-testid="display_title">{title}</ModelTitle>
                <ModelContent data-testid="display_message">{message}</ModelContent>
                <ModelFooter>
                    {
                        cancel ? (<Button type="cancel" onClick={cancel} data-testid="display_cancel_btn">{cancelText}</Button>) : null
                    }
                    {
                        confirm ? (<Button type="confirm" onClick={confirm} data-testid="display_confirm_btn">{confirmText}</Button>) : null
                    }
                </ModelFooter>
            </ModelWrap>
            <ModelShadow />
        </div>
    );

    return {
        ModelBox,
        isShown,
        showModal,
        hideModal
    };
};

export default useModel;
