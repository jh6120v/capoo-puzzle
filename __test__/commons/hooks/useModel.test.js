import React from 'react';
import { render } from '@testing-library/react';
import { renderHook, act } from '@testing-library/react-hooks';
import useModel from '../../../src/commons/hooks/useModel';

describe('test useModel hook', () => {
    it('test', () => {
        const { result } = renderHook(() => useModel('title', 'message', () => {

        }, 'Confirm', () => {

        }, 'Cancel'));

        const { container, getByTestId } = render(<result.current.ModelBox />);

        expect(container.firstChild).toMatchSnapshot();

        expect(getByTestId('display_title').innerHTML).toEqual('title');
        expect(getByTestId('display_message').innerHTML).toEqual('message');
        expect(getByTestId('display_confirm_btn').innerHTML).toEqual('Confirm');
        expect(getByTestId('display_cancel_btn').innerHTML).toEqual('Cancel');

        act(() => {
            result.current.showModal();
        });

        expect(result.current.isShown).toBe(true);

        act(() => {
            result.current.hideModal();
        });

        expect(result.current.isShown).toBe(false);
    });
});
