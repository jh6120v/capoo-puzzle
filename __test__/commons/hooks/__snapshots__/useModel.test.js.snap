// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test useModel hook test 1`] = `
<div>
  <div
    class="sc-bwzfXH ebVDIy"
  >
    <div
      class="sc-htpNat uEuno"
      data-testid="display_title"
    >
      title
    </div>
    <div
      class="sc-bxivhb iCCdRw"
      data-testid="display_message"
    >
      message
    </div>
    <div
      class="sc-ifAKCX kdvBBw"
    >
      <div
        class="sc-EHOje kNhJuu"
        data-testid="display_cancel_btn"
        type="cancel"
      >
        Cancel
      </div>
      <div
        class="sc-EHOje brQrel"
        data-testid="display_confirm_btn"
        type="confirm"
      >
        Confirm
      </div>
    </div>
  </div>
  <div
    class="sc-bdVaJa giODjc"
  />
</div>
`;
